_protected/app/system/modules/affiliate/forms/processing/LoginFormProcess.php

```diff
@@ -126,8 +126,8 @@ public function __construct()
      */
     public function updatePwdHashIfNeeded(string $sPassword, string $sUserPasswordHash, string $sEmail): void
     {
-        if ($sNewPwdHash = Security::pwdNeedsRehash($sPassword, $sUserPasswordHash)) {
-            $this->oAffModel->changePassword($sEmail, $sNewPwdHash, DbTableName::AFFILIATE);
+        if (Security::pwdNeedsRehash($sPassword, $sUserPasswordHash)) {
+            $this->oAffModel->changePassword($sEmail, $sPassword, DbTableName::AFFILIATE);
```

Comment:
Same here (and below). This won't work if we update the new password to the database with the new "clear" password. The current version should work. I've already looked at your issue last year and the occurred issue shouldn't have happened. I am aware that newer PHP versions can have breaking changes, (e.g. with the native password hashing API, `Security:: hashPwd`, `Security::pwdNeedsRehash`, ...), however the current version of pH7Builder should remain compatible. Your issue may have been caused by something else then. Thank you anyway for looking into it! I really appreciate your help here!! Always more than welcome!

Reply:
I know that we save password in hashed database. I have already explained the issue in detail in my first comment. You can also test it on your own, read the parts of the code, follow the logic.

This function saves the password on database, it should save the hashed password OF the clear password. NOT hashed password of the already hashed password. The function needs the clear password to be hashed, then save that hashed password. But it sends a hashed password to it, which causes the mentioned errors. Instead of updating the hash every time user logs in, it updates and changes the password every time user logs in. That means the user can't login the second time because the password is changed.


This dating Dating app system is far from compatible; after fixing that bug I discovered many other bugs, so I just stopped working with that.